src/app/(shared-nav)/courses/page.tsx

```diff
@@ -0,0 +1,104 @@
+"use client"
+import { languages } from "@/enums/language.enum";
+import { levels } from "@/enums/level.enum";
+import { CourseCard } from "@/components/courseCard";
+import { courseDetails } from "@/types/courseDetails.type";
+import { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { IoSearch } from "react-icons/io5";
+import {v4 as uuidv4} from "uuid";
+
+export default function Courses(){
+    const [searchText, setSearchText] = useState<String>("");
+    const [courses, setCourses] = useState<courseDetails[]>(DATA);
+    const handleSearch = (event: ChangeEvent<HTMLInputElement>)=>{
+        setSearchText(event.target.value.trim());
+    }
+    useEffect(()=>{
+        if(searchText)
+            setCourses(DATA.filter(course=> course.title.includes(searchText as string)))
+        else{
+            setSearchText("");
+            setCourses(DATA);
+        }
+    }, [searchText]);
+    const handleSuggestion = useCallback((title: string)=>{
```

Comment:
@haythamhakim can you please verify this solution for your [last comment ](https://github.com/Warriors-of-Light/sahwa/pull/1#discussion_r1647129138)?